### ISSUE TITLE
Placeholders to pass values to template rendering

### ISSUE THREAD

**Issue by aliok** [knative/func#1996]
When I create a NodeJS HTTP func using the template, I see it is always created with the same package name: `http-handler`: https://github.com/knative/func/blob/main/templates/node/http/package.json#L2

```shell
kn func create -l node -t http
```

It would be useful to make some[^1] of these hardcoded values placeholders.

Something like this:

```shell
kn func create -l node -t http -p name:"my-handler"
```

which can generate something like:

```...
{
  "name": "my-handler",
  "version": "0.1.0",
```

Current hardcoded values can be the defaults.

While `name` is just very simple and doesn't provide much value, having ports, builders, baseImages, etc. can be very useful.

### Context:
- I am working on creating some Backstage templates for Knative Function templates.
- If there was a way to pass some arguments to the function during the creation, I wouldn't need to duplicate these templates and add some placeholders in them

[^1]: Definition of what to keep hardcoded and what to parametrize needs some work.

**Comment by aliok** [comment]
cc @matzew

**Comment by matzew** [comment]
yes this would be super helpful

**Comment by aliok** [comment]
@matzew  @lkingland What do you think?

**Comment by lkingland** [comment]
> @matzew  @lkingland What do you think?

Yes, I would like to explore offering more customizable templates and integration with Backstage.

Let's discuss how this might work in practice either async via Slack or a meeting; at your convenience. I have a few thoughts.
